core/ts.core/src/ts/utils/ZipUtils.java

```diff
@@ -166,9 +168,13 @@ public static void extractTar(File file, File destination) throws IOException {
 
 					// Close the stream
 					out.close();
-					if (extractedFile.getParent().contains(BIN_FOLDER)) {
-						extractedFile.setExecutable(true);
-					}
+					// Preserve original modification date
+					extractedFile.setLastModified(entry.getTime());
+					long mode = entry.getMode(); 
+					if ((mode & 00100) > 0) { 
+						// Preserve execute permissions 
+						extractedFile.setExecutable(true, (mode & 00001) == 0); 
+					} 
 					break;
 				case TarEntry.LINK:
 					File linkFile = new File(destination, outFilename);
```

Comment:
Ok thanks for your feedback. If it works for you, it's OK.

> Have you thought on using Apache Commons Compress to avoid this low level implementation? 

I would like to avoid having dependencies just to extract node.js. We will see in the future if we have so many problem, perhaps I will switch to appache commons compress.